Bring quickstart to the fore and simplify (#5696)
treeverse/lakeFS: README.md

<p align="center">
  <img src="docs/assets/img/logo_large.png"/>
</p>

>  

Reviewer: Nit: probably unintended whitespace diff.
```suggestion

```

lakeFS supports AWS S3, Azure Blob Storage, and Google Cloud Storage as its underlying storage service. It is API compatible with S3 and works seamlessly with all modern data frameworks such as Spark, Hive, AWS Athena, DuckDB, and Presto.

For more information, see the [documentation](https://docs.lakefs.io).

## Getting Started

> You can spin up a standalone sandbox instance of lakeFS quickly under Docker by running the following:

Reviewer 2: ```suggestion
You can spin up a standalone sandbox instance of lakeFS using Docker:
```

Rolling back returns data to a state in the past, before the error was present. You might not be showing all the latest data after a rollback, but at least you aren’t showing incorrect data or raising errors. Since lakeFS provides versions of the data without making copies of the data, you can time travel between versions and roll back to the version of the data before the error was presented.

## Getting Started

You can spin up a standalone sandbox instance of lakeFS quickly under Docker by running the following:

```bash
docker run --pull always \
		   --name lakefs \
		   -p 8000:8000 \
		   treeverse/lakefs \
		   	 run --local-settings
```

### Other quickstart methods

> If you don't want to use Docker you can [install and run lakeFS locally](https://docs.lakefs.io/quickstart/more_quickstart_options.html#using-the-binary).

Reviewer: It is really unfortunate to lose this (I couldn't find it on https://docs.lakefs.io/quickstart, and it seems https://docs.lakefs.io/quickstart/more_quickstart_options.html is now gone!).  My concern is that while it might make life easier for lakeFS _users_ but much harder for lakeFS _developers_.  Developers probably need a way to run lakeFS with more knobs than the Docker image.  And GitHub is for developers.

I'm fine with having a more appealing README, but I would like it to be usable by developers.  It is already much too hard to onboard new developers -- not having any instructions in the README makes it even harder.

Author: This is a great point, thanks @arielshaqed. 

It then prompts the question (real, not rhetorical): should we have one README targeting both audiences (different sections of the same doc to cover different areas) or a separate README for developers? WDYT? 

I'm also interested in notes & comments you've got regarding what else we can do to address your point that _"it is already much too hard to onboard new developers"_

Reviewer 3: Think we should keep a very basic quick start that is usable as part of the project's README - it doesn't need to be 1:1 to the quick start as it will enables step by step explanations.

Author: ack. I've added this back in - please re-review @nopcoder @arielshaqed
